crates/sui-indexer-alt-framework/src/pipeline/concurrent/mod.rs

```diff
@@ -218,12 +228,29 @@ pub(crate) fn pipeline<H: Handler + Send + Sync + 'static>(
     let pruner_cancel = cancel.child_token();
     let handler = Arc::new(handler);
 
+    // If the watermark key does not match the pipeline name, then it is a tasked pipeline and we
+    // need to track the main reader lo.
+    let main_reader_lo = if !watermark_key.eq(H::NAME) {
```

Comment:
This logic would also be neater/less hacky if it was the optional task name that was passed in here, and not the pre-computed key.